src/tensor_ops.jl

```diff
@@ -152,17 +157,22 @@ Computes the trace of a second or fourth order tensor.
     @code :(return s)
 end
 
+const vol = trace
+
+
 ########
 # Mean #
 ########
+
 Base.mean{dim}(S::SecondOrderTensor{dim}) = trace(S) / dim
```

Comment:
If you don't specify `dim` then any `dim` will be ok for this function which is what we want. It is only when you save things in `type` and `immutable` you have to be careful about not storing abstract types.